### PR TITLE
Add changes for next version 2

### PR DESCRIPTION
**Pull request by arunoda**
Now uses custom routing to use to show the modal.
Here now it uses a single page to show and list photos.
We use custom routing to change the URL.

Here's the deployed URL: https://nextgram2.now.sh/

If needed, for the direct photo URL, we can get rid of the modal and just show the photo in a frame.
But I'm not sure, that's the functionality we need.

### CODE REVIEW

File: pages/photo.js [hunk 1]
```diff
@@ -1,24 +0,0 @@
...
-    padding: '100px',
-    textAlign: 'center'
-  },
-
-  wrap: {
-    display: 'inline-block',
-    border: '1px solid #999',
-    margin: 'auto'
-  }
-}
```
Comment (nkzawa) [inline pages/photo.js]: I think we shouldn't remove this.

Comment (nkzawa) [inline pages/photo.js]: As an example, I think we'd like to show how to render a modal with `url.push` and show a normal page on SSR.

Comment (arunoda) [inline pages/photo.js]: That's done in the `index.js` already. If we need to keep this, we need to serve this page with a different URL.

Comment (nkzawa) [inline pages/photo.js]: Basically, we don't want to have the photo detail on `index.js` (which a page for list), and showing the modal should be only the special case on this example.

Comment (arunoda) [inline pages/photo.js]: In that case, we need to use a different page name. 
We have already have assigned /photo to index.js via custom routing.

File: server.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      app.render(req, res, '/', { photoId: req.params.id })
+    })
+
+    server.get('*', handle)
+
+    console.log(`Nextgram started on port: ${port}`)
+    server.listen(port)
+  })
+  .catch((err) => {
+    throw err
```
Comment (nkzawa) [inline server.js]: You can't throw errors on `catch` since it's caught by Promise again.

Comment (arunoda) [inline server.js]: Yep.

File: pages/photo.js (hunk 1 above)
Comment (arunoda) [inline pages/photo.js]: @nkzawa updated as per your suggestion.
We can have /photo/:id like routing with the custom router API. Do we need to do it?

Comment (nkzawa) [inline pages/photo.js]: @arunoda I think `/photo/:id` is awesome, but can be a different PR.
It seems we have to recreate `pages/photo.js`.

Comment (arunoda) [inline pages/photo.js]: You mean?

Comment (nkzawa) [inline pages/photo.js]: Please add `pages/photo.js`. otherwise it's 404 on SSR.

Comment (arunoda) [inline pages/photo.js]: Oh my! Forgot to add it back to git. Done.

File: package.json [hunk 3]
```diff
@@ -2,17 +2,20 @@
...
   "dependencies": {
-    "next": "latest",
-    "standard": "8.4.0",
-    "babel-eslint": "7.0.0"
+    "babel-eslint": "7.0.0",
+    "express": "^4.14.0",
+    "global": "^4.3.1",
+    "next": "^2.0.0-beta",
+    "now": "^2.0.3",
+    "standard": "8.4.0"
```
Comment (nkzawa) [inline package.json]: where did `global` and `now` come from ?

Comment (arunoda) [inline package.json]: Removed.
